### PR TITLE
feat(ESSNTL-3576): introduces system update method filter

### PR DESCRIPTION
**Pull request by mkholjuraev**
Resolves: [ESSNTL-3576](https://issues.redhat.com/browse/ESSNTL-3576)

This introduces a new 'System update method' filter in the inventory table.

To test:

1. Visit the inventory page
2. Observe that the new filter is added to the filters dropdown
3. The filter works as expected. Sends a request as described in the ticket  (e.g `filter[system_profile][system_update_method][eq]=yum`).
4. Observe that the filter persists across refresh

### CODE REVIEW

File: src/components/filters/useUpdateMethod.test.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/* eslint-disable react/prop-types */
+import React, { Fragment, useEffect } from 'react';
+import { useUpdateMethodFilter } from './useUpdateMethodFilter';
+import { updateMethodOptions, UPDATE_METHOD_KEY } from '../../Utilities';
+import { mount } from 'enzyme';
```
Comment (bastilian) [inline src/components/filters/useUpdateMethod.test.js:5]: We should avoid `enzyme`. The `@testing-library/react-hooks` is already included in inventory and can be easily used here.

Comment (mkholjuraev) [reply src/components/filters/useUpdateMethod.test.js:5]: @bastilian thank you for having a look. Actually, `enzyme` is used in almost all filter tests, thus I prefer to get rid of it at once later. I would like to refactor the filtering in Inventory, while refactoring I will get rid of enzyme.

Comment (mkholjuraev) [reply src/components/filters/useUpdateMethod.test.js:5]: There is already a ticket to optimise filtering: [RHIF-112](https://issues.redhat.com/browse/RHIF-112)

**Review by bastilian — APPROVED** [review]
Works as expected! Thank you @mkholjuraev!

**Pull request merged**